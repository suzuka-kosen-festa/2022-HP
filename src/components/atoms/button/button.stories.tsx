import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Button } from ".";

type T = typeof Button;
type Story = ComponentStoryObj<T>

const data = {
  sentence: "This is test",
};

export default { args: { children: data.sentence }, component: Button, title: "Button" } as ComponentMeta<T>;

export const Default: Story = {};

export const LinkButton: Story = {
  args: {
    as: "a",
    role: "link"
  }
}
