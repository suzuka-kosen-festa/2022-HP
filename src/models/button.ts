import type { CSSInterpolation } from "@emotion/serialize";

export type ButtonProperties<T extends AnyComponent> = React.PropsWithChildren<
  PropertiesOf<T> & {
    boxStyles?: CSSInterpolation;
    textStyles?: CSSInterpolation;
    ref?: never;
    as?: T;
  }
>;
