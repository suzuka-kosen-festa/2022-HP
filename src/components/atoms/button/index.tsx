import React from "react";
import tw from "twin.macro";
import { ButtonProperties } from "../../../models";
import { Text } from "../text";

const ButtonBox = tw.button`bg-white hover:bg-button-hover flex flex-row items-start px-8 py-2 gap-2.5 rounded-[2rem]`;

const Button = <T extends AnyComponent>({
  children,
  boxStyles,
  textStyles,
  as,
  ...rest
}: ButtonProperties<T>): JSX.Element => (
  <ButtonBox css={boxStyles} as={as} {...rest}>
    <Text css={[tw`text-xs font-bold text-black`, textStyles]}>{children}</Text>
  </ButtonBox>
);

export { Button };
