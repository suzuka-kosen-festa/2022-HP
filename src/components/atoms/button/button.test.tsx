import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./button.stories";

const { Default, AnchorButton } = composeStories(stories);

const options = { name: "This is test" };

describe("(components) atoms/button", () => {
  test("default button should be atom", () => {
    const { container } = render(<Default />);
    expect(container).toBeAtom();
  });
  test("to be [role=button]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("button", options)).toBeInTheDocument();
  });
  test("tobe [role=button]", () => {
    const { getByRole } = render(<AnchorButton />);
    expect(getByRole("button", options)).toBeInTheDocument();
  });
  test("take default button's snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
  test("take anchor button's snap shot", () => {
    const { container } = render(<AnchorButton />);
    expect(container).toMatchSnapshot();
  });
});
