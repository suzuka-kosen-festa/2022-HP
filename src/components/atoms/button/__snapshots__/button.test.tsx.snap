// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(components) atoms/button take anchor button's snap shot 1`] = `
<div>
  <a
    class="twin-1fs6bgp-ButtonBox elv5i9d0"
    href="#"
    role="button"
  >
    <p
      class="twin-1u06aab-Text-Button epv4mo00"
    >
      This is test
    </p>
  </a>
</div>
`;

exports[`(components) atoms/button take default button's snap shot 1`] = `
<div>
  <button
    class="twin-1fs6bgp-ButtonBox elv5i9d0"
  >
    <p
      class="twin-1u06aab-Text-Button epv4mo00"
    >
      This is test
    </p>
  </button>
</div>
`;
